AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/CreateKeys.dfy

```diff
@@ -330,6 +349,27 @@ module {:options "/functionSyntax:4" } CreateKeys {
     var activeBranchKeyContext := Structure.ActiveBranchKeyEncryptionContext(decryptOnlyBranchKeyContext);
     var beaconBranchKeyContext := Structure.BeaconKeyEncryptionContext(decryptOnlyBranchKeyContext);
 
+    if !HvUtils.HasUniqueTransformedKeys?(activeBranchKeyContext) {
+      return Failure(Types.BranchKeyCiphertextException(
+                       message := ErrorMessages.NOT_UNIQUE_BRANCH_KEY_CONTEXT_KEYS
+                     ));
+    }
+    var ecToKMS := HvUtils.SelectKmsEncryptionContextForHv2(activeBranchKeyContext);
+
+    // get plaintext data key by calling kms::GenerateDataKey
+    var branchKey :- KMSKeystoreOperations.GetPlaintextDataKeyViaGenerateDataKey(
```

Comment:
~~Strong Suggestion, though not quite blocking~~ Blocking/Suggestion:
It has been very, very, difficult to explain that a Branch Key is composed of 3 or more items, and that at least 2 of these items have the same material.

Explaining that a **"Branch Key is composed of 3 or more items, and that at least 2 of these items have the same Branch Key"** has been an act of futility that I have stopped doing because two things that are different but very adjacent cannot have the same name without causing confusion.
```suggestion
    var activePlaintextMaterial :- KMSKeystoreOperations.GetPlaintextDataKeyViaGenerateDataKey(
```